File: services/backend/plan_service/src/main/java/com/example/plan_service/business/interfaces/PlanServiceInterface.java
package com.example.plan_service.business.interfaces;

import com.example.plan_service.presentation.http.Response;
import com.example.plan_service.presentation.requests.UserPlanRequest;

public interface PlanServiceInterface {
    Response<Object> getPlan(Integer idUser);
    Response<Object> deletePlan(Integer idUser);
    Response<Object> createUserPlan(Integer idUser, UserPlanRequest data);
}

File: services/backend/orchestrator_service/src/main/kotlin/com/example/orchestrator_service/business/interfaces/PlanServiceInterface.kt
package com.example.orchestrator_service.business.interfaces

import com.example.orchestrator_service.business.models.plan.request.CreateUserPlanRequest
import com.example.orchestrator_service.business.models.plan.response.PlanModelResponse
import com.example.orchestrator_service.presentation.http.Response

interface PlanServiceInterface {
    suspend fun getUserPlan(token: String): Response<PlanModelResponse>
    suspend fun createUserPlan(data: CreateUserPlanRequest, token: String): Response<PlanModelResponse>
    suspend fun deleteUserPlan(token: String): Response<out Any>
}